src/private/authorizePrivateHandler.ts

```diff
@@ -0,0 +1,50 @@
+import { jsonCodec, PrivateNatsHandler } from "../nats/nats";
+import { Subscription } from "nats";
+import { HANDLERS_SUBJECTS } from "../config";
+import {
+    AuthorizeInterface,
+    AuthorizeResponseInterface
+} from "../interfaces/authorize";
+import { authorizeService } from "../di.config";
+import AuthorizeCommand from "../commands/authorize/authorizeCommand";
+
+export const authorizePrivateHandlers: PrivateNatsHandler[] = [
+    [
+        HANDLERS_SUBJECTS.AUTHORIZE,
+        async (subscription: Subscription): Promise<void> => {
+            for await (const message of subscription) {
+                try {
+                    const { headers } = message;
+                    const data = jsonCodec.decode(
+                        message.data
+                    ) as AuthorizeInterface;
```

Comment:
same here. Let's use the generic `deserialize` function to `jsonCodec.decode` payloads and assign them a type.